docs/tutorials/basic/web.md

```diff
@@ -0,0 +1,219 @@
+---
+bodyclass: docs
+layout: docs
+headline: gRPC Basics - Web
+sidenav: doc-side-tutorial-nav.html
+type: markdown
+---
+<p class="lead">This tutorial provides a basic introduction on how to use
+gRPC-Web from browsers.</p>
+
+By walking through this example you'll learn how to:
+
+- Define a service in a .proto file.
+- Generate client code using the protocol buffer compiler.
+- Use the gRPC-Web API to write a simple client for your service.
+
+It assumes a passing familiarity with [protocol
+buffers](https://developers.google.com/protocol-buffers/docs/overview).
+
+<div id="toc"></div>
+
+<a name="why-grpc"></a>
+
+## Why use gRPC?
+
+With gRPC you can define your service once in a .proto file and implement
+clients and servers in any of gRPC's supported languages, which in turn can be
+run in environments ranging from servers inside Google to your own tablet - all
+the complexity of communication between different languages and environments is
+handled for you by gRPC. You also get all the advantages of working with
+protocol buffers, including efficient serialization, a simple IDL, and easy
+interface updating.
+
+
+<a name="setup"></a>
+
+## Define the Service
+
+The first step when creating a gRPC service is to define the service methods
+and their request and response message types using protocol buffers. In this
+example, we define our `EchoService` in a file called
+[`echo.proto`](https://github.com/grpc/grpc-web/blob/master/net/grpc/gateway/examples/echo/echo.proto).
+For more information about protocol buffers and proto3 syntax, please see the
+[protobuf documentation][].
+
+```protobuf
+message EchoRequest {
+  string message = 1;
+}
+
+message EchoResponse {
+  string message = 1;
+}
+
+service EchoService {
+  rpc Echo(EchoRequest) returns (EchoResponse);
+}
+```
+
+## Implement gRPC Backend Server
```

Comment:
This was a point of confusion for me personally when looking for web examples, I don't work with c++, I have to install dependencies and learn how to compile this server to run the test, it would be great to have a nodejs example for the server; I would assume that most people looking at the web examples are familiar with javascript, even if they dont know nodejs they'll be able to reason about it well enough.

Reply:
Agreed with this point from @gregkeys. Let's also add a NodeJS example. 
Should be straightforward to copy from https://github.com/grpc/grpc/blob/master/examples/node/dynamic_codegen/greeter_server.js 

@stanley-cheung should I send a PR to your PR repo to add a NodeJS example in this way?

Reply:
@hsaliak, yes that will be great

Reply:
Fixed. Added a Node backend gRPC server example.